veps/sig-compute/tdx-sev-snp-proposal.md

```diff
@@ -0,0 +1,236 @@
+# VEP #80: Enable TDX and SEV-SNP
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+This proposal deals with the enablement of the following technologies: Intel
+TDX & AMD Secure Encrypted Virtualization with Secure Nested Paging (SEV-SNP)
+in kubevirt. This enablement includes the support for creating confidential VMs
+by using TDX & SEV-SNP by encrypting VM memory. For AMD, SEV is currently
+supported in KubeVirt, and this VEP proposes to build on top of this feature
+and integrates SEV-SNP support.
+
+## Motivation
+Organizations are now asking for workloads to include the strongest
+confidential computing security guarantees.
+
+For AMD, KubeVirt supports SEV and SEV-ES for memory encryption, but lacks
+SEV-SNP enhanced memory integrity protection.
+
+For Intel, Add support for TDX technology for in-use memory encryption of
+confidential guests.
+
+## Goals
+- Enable TDX VM Deployment: Provide users the ability to deploy VMs that are
+  utilizing INTEL TDX technology.
+- Enable SEV-SNP VM Deployment: Provide users the ability to deploy VMs that
+  are utilizing AMD SEV-SNP technology.
+- Extend the existing SEV Feature: Currently the SEV sits behind the
+  WorkloadEncryptionSEV adding SEV-SNP to feature is the next logical step.
+- Extend Existing SEV Infrastructure: Use the existing SEV node labeling,
+  scheduling, and domain generation infrastructure.
+
+## Non Goals
+- TDX & SEV-SNP Live Migration: Live migrations are currently not supported by
+  either technologies.
```

Comment:
```suggestion
  either technologies. Goal: mark confidential-computing-enabled VMs as non-migratable.
```

Reply:
Either way, it's kind of a goal about a non-goal :) Just though it was worth mentioning somewhere.

Reply:
This is just to make sure the implementation prevents VMs from migrating, instead of letting them try and fail.